internal/ethapi/api.go

```diff
@@ -1262,6 +1262,25 @@ func (s *PublicBlockChainAPI) EstimateGas(ctx context.Context, args CallArgs, bl
 
 		return true, nil, nil, nil
 	}
+
+	// If the transaction is a plain value transfer, short circuit estimation and
+	// directly try 21000. Returning 21000 without any execution is dangerous as
+	// some tx field combos might bump the price up even for plain transfers (e.g.
+	// unused access list items). Ever so slightly wasteful, but safer overall.
+	if len(args.Data) == 0 && args.To != nil {
+		statedb, _, err := s.b.StateAndHeaderByNumber(ctx, rpc.LatestBlockNumber)
+		if statedb == nil || err != nil {
+			return 0, err
+		}
+
+		if statedb.GetCodeSize(*args.To) == 0 {
+			ok, _, err, _ := executable(params.TxGas)
```

Comment:
handle err != nil

Reply:
If `err != nil`, it maybe caused by insufficient gas or other reason although it rarely appears. The subsequent code will continue to handle this situation.